components/camel-spring/src/main/java/org/apache/camel/spring/SpringCamelContext.java

```diff
@@ -266,39 +233,58 @@ protected ModelJAXBContextFactory createModelJAXBContextFactory() {
         return new SpringModelJAXBContextFactory();
     }
 
-    private void maybeStart() throws Exception {
-        // for example from unit testing we want to start Camel later and not when Spring framework
-        // publish a ContextRefreshedEvent
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        sb.append("SpringCamelContext(").append(getName()).append(")");
+        if (applicationContext != null) {
+            sb.append(" with spring id ").append(applicationContext.getId());
+        }
+        return sb.toString();
+    }
+
+    @Override
+    public int getPhase() {
```

Comment:
I think I need to do more tests, I did this in a TDD style and arrived here, but I most likely have missed some scenarios. Having those different is a smell...